### DISCUSSION TITLE
Help required to update cQube_Workflow config.yml

### DISCUSSION THREAD

**Discussion by srajasimman** [Sunbird-cQube/community#33] [Q&A] [answered]
Hi @SreenivasNimmagadda,

Kindly help us to fill the following values in the config.yml to proceed with the cQube_Workflow installation.
```
---
base_dir: /opt                            # provide the absolute path where cqube needs to be installed and served. Eg. /opt  ( in this case cqube will be installed under /opt/cqube)
state_code: GJ                              # Enter the 2 character (upper case) state code for which cQube is being deployed. Please refer state_list file. 
diksha_columns: false                    # Enter true or false. 
static_datasource: state                  # Enter udise or state
management:                              # Enter the management
session_timeout: 7D                      # enter the value between 30 Minutes to 3650 Days. Eg. For minutes `60M`  Eg. For days `15D`. Make sure M and D should be mentioned upper case.
map_name: leafletmap                                # Enter the map_name ( mapmyindia or googlemap or leafletmap )
google_api_key: remove_this_key_value                        # Enter the google_api_key  here (if you selected map_name as googlemap replace `remove_this_key_value` with the actual googlemaps api key )
 
theme:                                    # Enter the theme ( theme1 or theme2 or theme3 )
slab1:                                     #Enter slab1 value between 1-100 ( eg: slab1: 33  )
slab2:                                     #Enter slab2 value range is greater than slab1 and between 1-100 ( eg: slab2: 33-60 ) 
slab3:                                     #Enter slab3 value range is greater than slab2 and between 1-100 ( eg: slab3: 60-75 ) 
slab4:                                     #Enter slab4 value is greater than or equal to slab3 last value between 1-100 ( eg: slab4 75 )
```

cc: @Ramya-kmurthy, @LavanyaKR3007, @gandham-santhosh

Thanks
Rajasimman S

**Comment by SreenivasNimmagadda** [comment, marked as answer]
@srajasimman,

Please find the below explanations and the sample values to fill the fields.

1. base_dir: "It is the Location of the cQube main directory for the installation"
2. state_code: "is the state code which represents the borders of the state in Map reports" 
   Please refer the available state codes in the [State_list](https://github.com/Sunbird-cQube/cQube_Workflow/blob/main/workflow_deploy/education_usecase/state_list)
3. diksha_columns: "This is for the Diksha stacked bar chart to enable or disable. Based on the data availability this report will be visible." In your case this value is 'false'
4. static_datasource: "This is for the static data either it is from UDISE or the state."
5. management: "This field will represent the default selection of the management. either it is govt schools, private, triable..etc "  make sure all the letters are in small case. for more understanding please find the below screenshot.
![image](https://user-images.githubusercontent.com/61133774/152971827-fb42765c-6f39-4185-93cc-358b367d98f3.png)
6. session_timeout: "This is for the report viewers session valid time. If the value is 7D, then the report viewer will not logged out for the next 7 days. after 7 days the session will be invalidate and the user will be logged out"
7. map_name: "This field is to represent the Map which you wanted to use. cQube is having the choices of Leaflet Maps, Google Maps and Map my India maps included."
8. google_api_key: "If you selected the Google maps in the above field, You have to enter the Google maps key in this field."
9. theme: "Theme is just to change the colors of the UI components. cQube is having 3 in-built themes Blue, Yellow and Green", You can create your custom theme and mark it as Theme 4 and mention it in this field if you want a new color codes."
10. Slabs: " Slabs are configuration values for the progress card. Progress card will show the development in 4 different slabs the slabs are like below 33% & 33% - 60% & 61% - 75% & Above 75% "

Hope all the points are clear for you. Please revert me back if you need more clarification on any point.

Thanks.
